ibis_substrait/compiler/translate.py

```diff
@@ -777,13 +777,32 @@ def selection(
 
     if selections := _get_selections(op):
 
-        if relation.project.common.ListFields():
-            # if there is already an `emit` in RelCommon then we're stacking
-            # projections and we need to update the output_mapping to refer to
-            # the number of fields present in the most recent emit
-            mapping_counter = itertools.count(
-                len(relation.project.common.emit.output_mapping)
+        rels = [
+            (attr, getattr(relation, attr))
+            for attr in (
+                "aggregate",
+                "cross",
+                "extension_leaf",
+                "extension_multi",
+                "extension_single",
+                "fetch",
+                "filter",
+                "join",
+                "project",
+                "read",
+                "set",
+                "sort",
             )
+        ]
+        for relname, rel in rels:
+            if relname == "aggregate" and rel.measures:
```

Comment:
Can you comment on why do we handle aggregate differently?

Reply:
see above in https://github.com/ibis-project/ibis-substrait/pull/466#issuecomment-1382311006